lib/my_tools/place_data_scraper.rb

```diff
@@ -22,8 +21,19 @@ def save_review(place_id)
     end
 
     def save_place
-      result = @scrape_process.fetch_place
-      Place.create(place_name: result[:place_name], address: result[:address])
+      uri = URI.parse(@url)
+      lrd = uri.fragment.split('=')[1].split(',')[0]
+      place = Place.find_by(lrd: lrd)
+      if place.present?
+        place
+      else
+        result = @scrape_process.fetch_place
+        Place.create(
+          lrd: lrd,
+          place_name: result[:place_name],
+          address: result[:address],
+        )
+      end
```

Comment:
このコード一緒に書いたの僕なのでこうコメントするのもなんですが……

```suggestion
      return place if place.present?
      result = @scrape_process.fetch_place
      Place.create(
        lrd: lrd,
        place_name: result[:place_name],
        address: result[:address],
      )
```

こんな感じで早期returnしてやって、ネスト（≒ インデント）減らすというテクニックをRubyではよく使いますね